core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -1062,6 +1062,25 @@ public void testJoinUsingDynamicTable() {
     sql("select * from dept, lateral table(ramp(deptno))").ok();
   }
 
+  @Test public void testSnapshotOnTemporalTable() {
+    final String sql = "select * from products_temporal "
+        + "for system_time as of TIMESTAMP '2011-01-02 00:00:00'";
+    sql(sql).ok();
+  }
+
+  @Test public void testJoinTemporalTableOnSpecificTime() {
+    final String sql = "select stream * from "
+        + "orders, products_temporal for system_time as of TIMESTAMP '2011-01-02 00:00:00'";
+    sql(sql).ok();
+  }
+
+  @Test public void testJoinTemporalTableOnColumnReference() {
+    final String sql = "select stream * from \n"
+        + "orders join products_temporal for system_time as of orders.rowtime \n"
```

Comment:
Hi @julianhyde  , the currently implementation of this PR doesn't need `LATERAL` keyword when FOR SYSTEM_TIME AS OF a left table's field. Do you think whether the LATERAL is needed?